barretenberg/cpp/src/barretenberg/commitment_schemes/commitment_key.hpp

```diff
@@ -78,8 +80,16 @@ template <class Curve> class CommitmentKey {
                  srs->get_monomial_size());
             ASSERT(false);
         }
-        return scalar_multiplication::pippenger_unsafe<Curve>(
-            const_cast<Fr*>(polynomial.data()), srs->get_monomial_points(), degree, pippenger_runtime_state);
+
+        // Extract the precomputed point table (contains raw SRS points at even indices and the corresponding
+        // endomorphism point (\beta*x, -y) at odd indices). We offset by polynomial.start_index * 2 to align
+        // with our polynomial span.
+        G1* point_table = srs->get_monomial_points() + polynomial.start_index * 2;
+        DEBUG_LOG_ALL(polynomial.span);
+        Commitment point = scalar_multiplication::pippenger_unsafe<Curve>(
+            polynomial.data(), point_table, degree, pippenger_runtime_state);
```

Comment:
Should actually update the name of the variable `degree` as it is now decidedly not the degree (and in fact never was)